pkg/utils/resources/resources.go

```diff
@@ -26,7 +26,7 @@ func RequestsForPods(pods ...*v1.Pod) v1.ResourceList {
 	var resources []v1.ResourceList
 	for _, pod := range pods {
 		for _, container := range pod.Spec.Containers {
-			resources = append(resources, container.Resources.Requests)
+			resources = append(resources, MergeResourceLimitsIntoRequests(container))
```

Comment:
I decided to merge limits into requests here to catch all scenarios where requests are not populated when limits are set (other than just DaemonSets).